gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegistrationUnit.java

```diff
@@ -124,13 +124,13 @@ protected void populateSerDeFields(State state) {
     if (state.contains(key)) {
       Optional<T> fieldValue;
 
-      if (new TypeToken<Boolean>() {}.isAssignableFrom(token)) {
+      if (new TypeToken<Boolean>() {}.isSubtypeOf(token)) {
```

Comment:
Check this doc https://github.com/google/guava/wiki/Release20. Check for the keyword isAssignableFrom